### PR TITLE
Lnprototest update, and timeout CI if it goes awry

### PR DESCRIPTION
**Pull request by rustyrussell**
Changelog-None

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -104,6 +104,7 @@ jobs:
       EXPERIMENTAL_FEATURES: 1
       COMPAT: 0
       PYTEST_PAR: 2
+      PYTEST_OPTS: "--timeout=300"
```
Comment (vincenzopalazzo) [inline .github/workflows/ci.yaml:107]: it is possible also to use the Github action timeout `timeout-minutes: 300` in order to make stay safe at all level

**Review by vincenzopalazzo — APPROVED** [review]
ack https://github.com/ElementsProject/lightning/pull/5125/commits/f4fae41c1b44ffdede646d835cab0b361b7b1b79

Good catch, I was looking into reproducing it but without luck!

**Comment by vincenzopalazzo** [comment]
Mh, I think this is a GitHub problem, the same lnprototest is raining on the docker image https://github.com/rustyrussell/lnprototest/runs/5670869353?check_suite_focus=true in 12 minutes including the c-lightning build!

I can make another PR with the migration to Docker? What do you think?

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (rustyrussell) [reply .github/workflows/ci.yaml:107]: No, I want this inside the test, so we get a backtrace...

Comment (vincenzopalazzo) [reply .github/workflows/ci.yaml:107]: Oh! I tested it and I got what you want!

I introduce in lnprototest also the logger that you can enable by passing to pytest `--log-cli-level=DEBUG`!

**Comment by vincenzopalazzo** [comment]
This PR should be full override with the https://github.com/ElementsProject/lightning/pull/5129 to stop the nondeterministic behavior from Github actions, and also with docker, we have the opportunity to run the same CI env locally!

**Comment by rustyrussell** [comment]
#5125  is better

**Pull request closed without merging**